src/fs.rs

```diff
@@ -50,14 +54,15 @@ fn set_file_contents(dirfd: &OwnedFd, name: &OsStr, stat: &Stat, data: &[u8]) ->
         }
         Err(Errno::OPNOTSUPP) => {
             // vfat? yolo...
-            let fd = openat(
+            let mut fd = openat(
                 dirfd,
                 name,
-                OFlags::CREATE | OFlags::WRONLY,
+                OFlags::CREATE | OFlags::WRONLY | OFlags::CLOEXEC,
                 stat.st_mode.into(),
-            )?;
-            write(&fd, data)?;
-            fdatasync(&fd)?;
+            )
+            .map(std::fs::File::from)?;
```

Comment:
Also: this is a bit DRY-unfriendly with respect to the block above.  I wonder if we should have some kind of a `util::write_to_fd()` helper that does this pattern of taking an `fd`, creating a `File` from it, `.write_all()` to it, then calling sync.  I know this is only strengthening your argument for `cap-std`... :)

Reply:
Another aside: I've been wondering if we should stop calling `fdatasync()` so much.  I've noticed that this is making things really slow.  I added `cache=unsafe` to our VM runs to help a bit, but it's clearly a problem.  We probably need to think about this more systematically...

In particular, do we really need to do this with fsverity files?  It probably makes sense there to let fsverity do its job, no?

This particular aside isn't actionable for this PR.  This code here is used for bootloader resources, so it's probably more valuable here...

Reply:
> Any reason to not just leave the ? on the openat() operation and do let file = File::from(fd); after?

I think that's a style thing that can go either way, will change it in the next push.

> I find that a lot more readable than having to know what .map() on an Option<> means.

This however: `map` on an Option is a very fundamental Rust thing I think (and it's not a complex API either). You *will* see nontrivial "combinator" usage out there (and to be fair it's easy to sometimes go too far on combinators) but this isn't at all unusual usage I think.

A basic example is:

```
let f = File::open("/etc/foo")?;
let f = BufReader::new(f);
```

vs

```
let f = File::open("/etc/foo").map(BufReader::new())?;
```

I use the latter a lot because I think it is just shorter and more readable.

Reply:
There's like 5 separate things in this one thread so not marking resolved but I changed the map

Reply:
Thanks.  The others were more like drive-bys for discussion points not directly related to anything here...

Reply:
> This however: `map` on an Option is a very fundamental Rust thing I think (and it's not a complex API either). You _will_ see nontrivial "combinator" usage out there (and to be fair it's easy to sometimes go too far on combinators) but this isn't at all unusual usage I think.

Ya.  I think this is maybe another case of me not being hip to the jive or whatever the kids these days are doing.  I know that I really dislike seeing `these.massive.chains().of().linked().operators()...`...

Reply:
> Ya. I think this is maybe another case of me not being hip to the jive or whatever the kids these days are doing. I know that I really dislike seeing these.massive.chains().of().linked().operators()......

There are IMO some things that are super elegant as combinators that are straightforward but verbose in explicit loop form, for example the `map()` + `.sum()` as used here:

https://github.com/containers/bootc/pull/921/files#diff-323d27aaf67b976dd83fc169a9fba2ea73f8d7aba9587cabce7f9bab9a804885R333